File: Api/EasyUi/src/models/nodes/node.ts
import React from "react";

export interface NodeDto {
    id: number;
    nodeName: string;
    sshHost: string;
    sshPort: number;
    sshUsername: string;
    method: number;
    serverPort: number;
    price: number;
    status: number;
    isAvailableForPurchase: boolean;
    xrayContainerImage: string | null;
    provisioningStatus?: string | number;
    provisioningMessage?: string | null;
    agentVersion?: string | null;
    targetAgentVersion?: string | null;
    lastSeenUtc?: string | null;
    installMethod?: string | number;
    marzbanEndpoint?: string | null;
    isEnabled?: boolean;
}

export interface InitiateNode {
    "nodeId": number;
    "panelId": number;
}
export type NodeStatus = 0 | 1 | 2;

export interface Props {
    node: { nodeName: string; price: number; status: NodeStatus };
    footer?: React.ReactNode;
    isConnected?: boolean;
    onSelect?: () => void;
    selecting?: boolean;
    className?: string;
}

File: Domain/Entities/Node.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;
using Domain.Enumes.Node;

namespace Domain.Entities;

[Table("Node", Schema = "Commerce")]
public class Node : BaseEntity<long>
{
    public string? NodeName { get; set; } 
    public int SshPort { get; set; } 
    public string? SshHost { get; set; } 
    public string? SshUsername { get; set; }
    public string? SshPassword { get; set; }
    public LoginMethod Method { get; set; } 

    public string SshKeyCertKey { get; set; } = string.Empty;
    public string? XrayContainerImage { get; set; } 
    public ICollection<Instance> Instances { get; set; } = new List<Instance>();
    public decimal Price { get; set; }
    public NodeStatus Status { get; set; } 
    public bool IsAvailableForShow { get; set; } 

    // --- New Agent / Provisioning Fields ---
    public bool IsEnabled { get; set; } = true; // drives auto-provisioning
    public NodeProvisioningStatus ProvisioningStatus { get; set; } = NodeProvisioningStatus.Pending;
    public string? ProvisioningMessage { get; set; }
    public string? AgentVersion { get; set; }
    public string? TargetAgentVersion { get; set; }
    public DateTime? LastSeenUtc { get; set; }
    public AgentInstallMethod InstallMethod { get; set; } = AgentInstallMethod.Docker;
    public string? MarzbanEndpoint { get; set; }
    public string? AgentEnrollmentToken { get; set; }
    public DateTime? AgentEnrollmentTokenExpiresUtc { get; set; }
}